.travis.yml

```diff
@@ -1,8 +1,8 @@
 language: go
 go:
- - 1.7
+ - 1.12
```

Comment:
Is 1.12 now the earliest version of go that we can build on? If so, we should update the README, but I'd prefer us to maintain compatibility with slightly older versions if possible.

Reply:
I am preparing for introducing go modules. 
this should also work with `1.7`  without any issues.   Only  `context` package will limit how far we can go with go versions (can't remember when it was introduced but it is not available in old versions)

#24 is the endgame. So I will be providing a series of patches and will update README accordingly.